neurom/core/_neuron.py

```diff
@@ -131,7 +206,7 @@ def transform(self, trans):
 
     def iter_sections(self, order=Tree.ipreorder):
         '''iteration over section nodes'''
-        return order(self.root_node)
+        return iter_sections(self, iterator_type=order)
```

Comment:
I am more than happy to have the iter_segments on the section since I opened #558